File: ipl-ecommerce-ui/src/app/models/order.ts
export interface OrderItem {
  id: number;
  productId: number;
  productName: string;
  quantity: number;
  unitPrice: number;
  totalPrice: number;
}

export interface Order {
  id: number;
  userId: number;
  orderDate: string;
  totalAmount: number;
  status: string;
  shippingAddress: string;
  items: OrderItem[];
}

export interface OrderSummary {
  id: number;
  orderDate: string;
  totalAmount: number;
  status: string;
  totalItems: number;
}

export interface CheckoutRequest {
  shippingAddress: string;
}

export interface CheckoutResponse {
  id: number;
}

File: Domain/Entities/Order.cs
using IPL.ECommerce.Domain.Entities;
using IPL.ECommerce.Domain.Enums;

public class Order
{
    public int Id { get; set; }

    public int UserId { get; set; }

    public DateTime OrderDate { get; set; } = DateTime.UtcNow;

    public decimal TotalAmount { get; set; }

    public OrderStatus Status { get; set; }

    public string ShippingAddress { get; set; } = string.Empty;

    public User User { get; set; } = null!;

    public ICollection<OrderItem> Items { get; set; }
        = new List<OrderItem>();
}